src/PowerShellEditorServices/Services/PowerShell/Debugging/DscBreakpointCapability.cs

```diff
@@ -0,0 +1,170 @@
+            // so store that for future use
+            if (breakpoints.Length > 0)
+            {
+                // Set the breakpoints for this scriptPath
+                this.breakpointsPerFile[scriptPath] =
+                    breakpoints.Select(b => b.LineNumber).ToArray();
+            }
+            else
+            {
+                // No more breakpoints for this scriptPath, remove it
+                this.breakpointsPerFile.Remove(scriptPath);
+            }
+
+            string hashtableString =
+                string.Join(
+                    ", ",
+                    this.breakpointsPerFile
+                        .Select(file => $"@{{Path=\"{file.Key}\";Line=@({string.Join(",", file.Value)})}}"));
+
+            // Run Enable-DscDebug as a script because running it as a PSCommand
+            // causes an error which states that the Breakpoint parameter has not
+            // been passed.
+            var dscCommand = new PSCommand().AddScript(
+                hashtableString.Length > 0
+                    ? $"Enable-DscDebug -Breakpoint {hashtableString}"
+                    : "Disable-DscDebug");
+
+            await executionService.ExecutePSCommandAsync(
+                dscCommand,
+                new PowerShellExecutionOptions(),
+                CancellationToken.None);
+
+            // Verify all the breakpoints and return them
+            foreach (var breakpoint in breakpoints)
+            {
+                breakpoint.Verified = true;
+            }
+
+            return breakpoints.ToArray();
+        }
+
+        public bool IsDscResourcePath(string scriptPath)
+        {
+            return dscResourceRootPaths.Any(
+                dscResourceRootPath =>
+                    scriptPath.StartsWith(
+                        dscResourceRootPath,
+                        StringComparison.CurrentCultureIgnoreCase));
+        }
+
+        public static async Task<DscBreakpointCapability> GetDscCapabilityAsync(
+            ILogger logger,
+            IRunspaceInfo currentRunspace,
+            PowerShellExecutionService executionService,
+            CancellationToken cancellationToken)
+        {
+            // DSC support is enabled only for Windows PowerShell.
+            if ((currentRunspace.PowerShellVersionDetails.Version.Major >= 6) &&
+                (currentRunspace.RunspaceOrigin != RunspaceOrigin.DebuggedRunspace))
+            {
+                return null;
+            }
+
+            Func<SMA.PowerShell, CancellationToken, DscBreakpointCapability> getDscBreakpointCapabilityFunc = (pwsh, cancellationToken) =>
+            {
+                PSModuleInfo dscModule = null;
+                try
+                {
+                    dscModule = pwsh.AddCommand("Import-Module")
+                        .AddArgument(@"C:\Program Files\DesiredStateConfiguration\1.0.0.0\Modules\PSDesiredStateConfiguration\PSDesiredStateConfiguration.psd1")
```

Comment:
😆

Reply:
@anmenaga do you know if this is how we should be looking for DSC?

Reply:
For 2 recent versions (v2.0.5 and v3) of DSC running in PS Core - I would rely on PSModulePath (`Get-Module -Name PSDesiredStateConfiguration -ListAvailable`)
But there is code above in this method that says "DSC support is enabled only for Windows PowerShell.".
In Windows PowerShell days it was quite possible to rely that module is in specific folder in Program Files.